src/libs/Navigation/Navigation.js

```diff
@@ -148,6 +148,70 @@ function setParams(params, routeKey) {
     });
 }
 
+/**
+ * Replace a given reportID in the navigation stack.
```

Comment:
This approach seems reasonable and sounds like it will give us the best possible UX, but _maybe_ could benefit from some more explanation or comments on how it all works. My concern is that we might miss a change in how the react navigation state works in the future and this will all break (but no real logical reason to give other than [this kind of comment from the doc](https://reactnavigation.org/docs/navigation-prop/#getstate)).

Reply:
We chatted about this a bit in person, but I feel like while having screens update their own params might be simpler, it also might create edge cases in UI – showing a 404 or the wrong report briefly before showing the correct report.

Reply:
Created a separate issue for the existing navigation bugs